src/wallet/wallet2.cpp

```diff
@@ -3404,6 +3594,209 @@ bool wallet2::load_tx(const std::string &signed_filename, std::vector<tools::wal
+  for (auto &ptx: exported_txs.m_ptx) LOG_PRINT_L0(cryptonote::obj_to_json_str(ptx.tx));
+
+  if (accept_func && !accept_func(exported_txs))
+  {
+    LOG_PRINT_L1("Transactions rejected by callback");
+    return false;
+  }
+
+  const bool is_signed = exported_txs.m_signers.size() >= m_multisig_threshold;
+  if (is_signed)
+  {
+    for (const auto &ptx: exported_txs.m_ptx)
+    {
+      const crypto::hash txid = get_transaction_hash(ptx.tx);
+      if (store_tx_info())
+      {
+        m_tx_keys.insert(std::make_pair(txid, ptx.tx_key));
+      }
+    }
+  }
+
+  return true;
+}
+//----------------------------------------------------------------------------------------------------
+bool wallet2::sign_multisig_tx(multisig_tx_set &exported_txs, const std::string &filename, std::vector<crypto::hash> &txids)
+{
+  THROW_WALLET_EXCEPTION_IF(exported_txs.m_ptx.empty(), error::wallet_internal_error, "No tx found");
+
+  // sign the transactions
+  for (size_t n = 0; n < exported_txs.m_ptx.size(); ++n)
+  {
+    tools::wallet2::pending_tx &ptx = exported_txs.m_ptx[n];
+    tools::wallet2::tx_construction_data &sd = ptx.construction_data;
+    LOG_PRINT_L1(" " << (n+1) << ": " << sd.sources.size() << " inputs, mixin " << (sd.sources[0].outputs.size()-1) <<
+        ", signed by " << exported_txs.m_signers.size() << "/" << m_multisig_threshold);
+    cryptonote::transaction tx;
+    std::vector<rct::multisig_out> msout;
+    bool r = cryptonote::construct_tx_with_tx_key(m_account.get_keys(), sd.sources, sd.splitted_dsts, sd.extra, tx, sd.unlock_time, ptx.tx_key, sd.use_rct, &msout);
+    THROW_WALLET_EXCEPTION_IF(!r, error::tx_not_constructed, sd.sources, sd.splitted_dsts, sd.unlock_time, m_testnet);
+
+    // since this is multisig, our local secret spend key will generate wrong key images,
+    // so we copy the incoming key images to the generated tx since this is expected
+    THROW_WALLET_EXCEPTION_IF(tx.vin.size() != ptx.tx.vin.size(),
+        error::wallet_internal_error, "Transaction prefix does not match data");
+    for (size_t i = 0; i < tx.vin.size(); ++i)
+    {
+      CHECKED_GET_SPECIFIC_VARIANT(tx.vin[i], txin_to_key, in, false);
+      CHECKED_GET_SPECIFIC_VARIANT(ptx.tx.vin[i], const txin_to_key, in_org, false);
+      in.k_image = in_org.k_image;
+    }
+
+    THROW_WALLET_EXCEPTION_IF(get_transaction_prefix_hash (tx) != get_transaction_prefix_hash(ptx.tx),
+        error::wallet_internal_error, "Transaction prefix does not match data");
+
+    // Tests passed, sign
+    rct::keyV k;
+    for (size_t idx: sd.selected_transfers)
+      k.push_back(get_multisig_k(idx));
+    std::vector<unsigned int> indices;
+    for (const auto &source: sd.sources)
+      indices.push_back(source.real_output);
+    THROW_WALLET_EXCEPTION_IF(!rct::signMultisig(ptx.tx.rct_signatures, indices, k, ptx.msout, rct::sk2rct(get_account().get_keys().m_spend_secret_key)),
+        error::wallet_internal_error, "Failed signing, transaction likely malformed");
```

Comment:
I added it to save_multisig_tx. It can't be in the transfer_selected_rct etc, as tx creation might fail, and/or the user might cancel a tx on prompt. This would mean the user can't redo another tx (using the same inputs). Having the zero in save_multisig_tx means the data is zeroed when the tx actually gets out. It's also in commit_tx.